File: src/main/java/hu/frontrider/arcana/containers/ContainerArcaneCage.java
package hu.frontrider.arcana.containers;

import hu.frontrider.arcana.blocks.experiments.tiles.TileEntityArcaneCage;
import net.minecraft.entity.player.EntityPlayer;
import net.minecraft.entity.player.InventoryPlayer;
import net.minecraftforge.items.CapabilityItemHandler;
import net.minecraftforge.items.IItemHandler;
import net.minecraftforge.items.SlotItemHandler;

public class ContainerArcaneCage extends ContainerBase {

    public ContainerArcaneCage(InventoryPlayer playerInv, final TileEntityArcaneCage tileEntityArcaneCage) {
        IItemHandler inventory = tileEntityArcaneCage.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null);

        this.addSlotToContainer(new SlotItemHandler(inventory, 0, 80, 35) {
            @Override
            public void onSlotChanged() {
                tileEntityArcaneCage.markDirty();
            }
        });

        this.addSlotToContainer(new SlotItemHandler(inventory, 1, 135, 35) {
            @Override
            public void onSlotChanged() {
                tileEntityArcaneCage.markDirty();
            }
        });


        placePlayerInv(playerInv);
    }


    @Override
    public boolean canInteractWith(EntityPlayer playerIn) {
        return true;
    }


}

File: src/main/java/hu/frontrider/arcana/content/containers/ContainerArcaneCage.kt
package hu.frontrider.arcana.content.containers

import hu.frontrider.arcana.content.blocks.experiments.tiles.TileEntityArcaneCage
import net.minecraft.entity.player.EntityPlayer
import net.minecraft.entity.player.InventoryPlayer
import net.minecraftforge.items.CapabilityItemHandler
import net.minecraftforge.items.IItemHandler
import net.minecraftforge.items.SlotItemHandler

class ContainerArcaneCage(playerInv: InventoryPlayer, tileEntityArcaneCage: TileEntityArcaneCage) : ContainerBase() {

    init {
        val inventory = tileEntityArcaneCage.getCapability(CapabilityItemHandler.ITEM_HANDLER_CAPABILITY, null)

        this.addSlotToContainer(object : SlotItemHandler(inventory, 0, 80, 35) {
            override fun onSlotChanged() {
                tileEntityArcaneCage.markDirty()
            }
        })

        this.addSlotToContainer(object : SlotItemHandler(inventory, 1, 135, 35) {
            override fun onSlotChanged() {
                tileEntityArcaneCage.markDirty()
            }
        })


        placePlayerInv(playerInv)
    }


    override fun canInteractWith(playerIn: EntityPlayer): Boolean {
        return true
    }


}
